dot/network/notifications.go

```diff
@@ -62,57 +59,67 @@ type handshakeReader struct {
 }
 
 type notificationsProtocol struct {
-	protocolID         protocol.ID
-	getHandshake       HandshakeGetter
-	handshakeDecoder   HandshakeDecoder
-	handshakeValidator HandshakeValidator
+	protocolID               protocol.ID
+	getHandshake             HandshakeGetter
+	handshakeDecoder         HandshakeDecoder
+	handshakeValidator       HandshakeValidator
+	outboundHandshakeMutexes *sync.Map //map[peer.ID]*sync.Mutex
+	inboundHandshakeData     *sync.Map //map[peer.ID]*handshakeData
+	outboundHandshakeData    *sync.Map //map[peer.ID]*handshakeData
+}
 
-	inboundHandshakeData  *sync.Map //map[peer.ID]*handshakeData
-	outboundHandshakeData *sync.Map //map[peer.ID]*handshakeData
+func newNotificationsProtocol(protocolID protocol.ID, handshakeGetter HandshakeGetter, handshakeDecoder HandshakeDecoder, handshakeValidator HandshakeValidator) *notificationsProtocol {
+	return &notificationsProtocol{
+		protocolID:               protocolID,
+		getHandshake:             handshakeGetter,
+		handshakeValidator:       handshakeValidator,
+		handshakeDecoder:         handshakeDecoder,
+		outboundHandshakeMutexes: new(sync.Map),
+		inboundHandshakeData:     new(sync.Map),
+		outboundHandshakeData:    new(sync.Map),
+	}
 }
 
-func (n *notificationsProtocol) getInboundHandshakeData(pid peer.ID) (handshakeData, bool) {
+func (n *notificationsProtocol) getInboundHandshakeData(pid peer.ID) (*handshakeData, bool) {
```

Comment:
nit maybe add named returned variables

```suggestion
func (n *notificationsProtocol) getInboundHandshakeData(pid peer.ID) (data *handshakeData, ok bool) {
```